acra-censor/handlers/blacklist_handler.go

```diff
@@ -204,22 +88,22 @@ func (handler *BlacklistHandler) Release() {
 // AddQueries normalizes and adds queries to the list that should be blacklisted
 func (handler *BlacklistHandler) AddQueries(queries []string) {
 	for _, query := range queries {
-		normalizedQuery, _, err := NormalizeAndRedactSQLQuery(query)
-		if err != nil {
-			continue
+		handler.queries[query] = true
+		normalizedQuery, _, _, err := common.HandleRawSQLQuery(query)
+		if err == nil {
+			handler.queries[normalizedQuery] = true
 		}
-		handler.queries[normalizedQuery] = true
```

Comment:
what about store query as is without normalization? to support unparsable queries

Reply:
oh, I missed `handler.queries[query] = true`
I think we should store one variant: normalized or as is. We don't need to store both IMHO

Reply:
I think that we should save both versions (normalized and raw). If parsing error occurred, we add/remove only raw version from list in all cases

Reply:
for what we need store both versions? how we use it? we can add/remove only normalized or raw query. anyway, we try to parse in both cases and check with logical OR both versions (in queryignore handler). so we can reduce overhead of memory and store one version

Reply:
In previous version we tried to parse and store normilized. If parse error occurred, we save as is. I agree that we have to save only normalized or raw. It seems that raw type is better - no parse, faster work

Reply:
Eventually I decided to use normalized queries CheckQuery and normalize them in Add/Remove Queries functions. So "SELECT * FROM STUDENTS;" will match "select * from students"

Reply:
if we can parse then we operate with parsed queries. and when we will try to add/remove parsable query then it will correctly added/removed. and when we cant parse then we add as is in raw form and add/remove will fail with parsing too and correctly added/removed using raw form of query

Reply:
I think we shouldn't allow add to black/white list unparsed queries - we check normalized queries in CheckQuery - not raw, and normalized query means that it's parsed.

Reply:
Currently every input query is parsed twice (one statement used for hiding values and second is processed) independently of further handling logic. When we parse while add/remove, it's done only once when loading. So even if we use unparsed queries in checking queries, It will not increase performance, but may cause some annoying typo errors, like add to whitelist "select * from x" but it will not block "select * from x;" because strings are not actually equal.